Tests occasionally (and seemingly randomly) fail on Travis

Failure messages:

    [Thu Apr 21 2016 21:01:14 GMT+0000 (UTC)] ERROR ::ffff:127.0.0.1 "/v0/projects/gwm": Error inserting slugs: Error: insert into "projectslugs" ("name", "project") values ('gwm', 6) - SQLITE_CONSTRAINT: UNIQUE constraint failed: projectslugs.name
    [Thu Apr 21 2016 21:01:14 GMT+0000 (UTC)] ERROR ::ffff:127.0.0.1 "/v0/projects/gwm": Rolling back transaction.
      1) successfully patches a project's uri, slugs, name, and default_activity by its manager

and:

    1) Endpoints POST /projects/:slug successfully patches a project's uri, slugs, name, and default_activity by its manager:
          Uncaught AssertionError: expected 500 to equal 200
          + expected - actual
          -500
          +200
          
          at Request._callback (tests/projects.js:454:37)
          at Request.self.callback (node_modules/request/request.js:200:22)
          at Request.<anonymous> (node_modules/request/request.js:1067:10)
          at IncomingMessage.<anonymous> (node_modules/request/request.js:988:12)
          at endReadableNT (_stream_readable.js:905:12)

At this time, I'm not sure what the cause is, but it doesn't seem to be indicative of any actual bugs in the application.

--- a/src/projects.js
+++ b/src/projects.js
@@ -750,23 +750,27 @@ module.exports = function(app) {
                               if (helpers.getType(obj.slugs) === 'array') {
                                 const newSlugs = [];
 
-                                newSlugs.push(trx('projectslugs').del()
-                                .where({project: oldId}));
+                                trx('projectslugs').del()
+                                .where({project: oldId}).then(function() {
+                                  /* eslint-disable */
+                                  for (let slug of obj.slugs) {
+                                  /* eslint-enable */
+                                    newSlugs.push(trx('projectslugs')
+                                    .insert({project: projID, name: slug}));
+                                  }
 
-                                /* eslint-disable */
-                                for (let slug of obj.slugs) {
-                                /* eslint-enable */
-                                  newSlugs.push(trx('projectslugs')
-                                  .insert({project: projID, name: slug}));
-                                }
-
-                                Promise.all(newSlugs).then(function() {
-                                  project.slugs = obj.slugs.sort();
-                                  delete project.newest;
-                                  trx.commit();
-                                  res.send(JSON.stringify(project));
+                                  Promise.all(newSlugs).then(function() {
+                                    project.slugs = obj.slugs.sort();
+                                    delete project.newest;
+                                    trx.commit();
+                                    res.send(JSON.stringify(project));
+                                  }).catch(function(error) {
+                                    log.error(req, 'Error inserting slugs: ' +
+                                                                        error);
+                                    trx.rollback();
+                                  });
                                 }).catch(function(error) {
-                                  log.error(req, 'Error inserting slugs: ' +
+                                  log.error(req, 'Error deleting old slugs: ' +
                                                                         error);
                                   trx.rollback();
                                 });
@@ -815,23 +819,28 @@ module.exports = function(app) {
                           if (helpers.getType(obj.slugs) === 'array') {
                             const newSlugs = [];
 
-                            newSlugs.push(trx('projectslugs').del()
-                            .where({project: oldId}));
+                            trx('projectslugs').del().where({project: oldId})
+                            .then(function() {
+                              /* eslint-disable */
+                              for (let slug of obj.slugs) {
+                              /* eslint-enable */
+                                newSlugs.push(trx('projectslugs')
+                                .insert({project: projID, name: slug}));
+                              }
 
-                            /* eslint-disable */
-                            for (let slug of obj.slugs) {
-                            /* eslint-enable */
-                              newSlugs.push(trx('projectslugs')
-                              .insert({project: projID, name: slug}));
-                            }
-
-                            Promise.all(newSlugs).then(function() {
-                              project.slugs = obj.slugs.sort();
-                              delete project.newest;
-                              trx.commit();
-                              res.send(JSON.stringify(project));
+                              Promise.all(newSlugs).then(function() {
+                                project.slugs = obj.slugs.sort();
+                                delete project.newest;
+                                trx.commit();
+                                res.send(JSON.stringify(project));
+                              }).catch(function(error) {
+                                log.error(req, 'Error inserting slugs: ' +
+                                          error);
+                                trx.rollback();
+                              });
                             }).catch(function(error) {
-                              log.error(req, 'Error inserting slugs: ' + error);
+                              log.error(req, 'Error deleting old slugs: ' +
+                                        error);
                               trx.rollback();
                             });
                           } else {

